### ISSUE TITLE
Simplify creation/management of communities

### ISSUE THREAD

**Issue by divnull** [friendica/friendica#6078]
A lot of soon-to-be-former G+ users are looking for a social media platform which has a feature the works like G+ communities do. Friendica can do this, via the community account type, but creation of management of them is unfriendly.

### Expected behavior

Some sort of a "create community" button someplace, which leads to a form where you set the name and preferences of the community, click create and be done.

### Actual behavior

Assuming you can figure out what you are actually supposed to do, you first log out. Create a new user for the community, going through the whole account creation process. Set the delegate for this new account to your main account. Log out. Log back into your main account.

### Possible Implementation

One way this might be accomplished is by doing something is with the notion of a "subordinate user" (or whatever you want to call it). One logged in user could create a subordinate user that:

- is automatically delegated to the creating user.
- has no password, and can never be used to log in. (If that isn't possible, you could emulate the same effect just by assigning the user a random password that no one ever sees and no e-mail address, though this would be less secure.)
- has some link back to the creator (unless the delegation is sufficient for this purpose).

Some mechanism would need to exist that would allow a user to transfer a "subordinate user" to another user.

**Comment by annando** [comment]
Technically this is no problem. Only thing is, that we need someone who builds a nice frontend for that. We already prepared some stuff, so that we could create communities (and other account types) from another user.

**Comment by divnull** [comment]
Isn't the UI just:

1. Make the "Manage" menu visible to everyone.
2. On the "Manage" page, below the icons of accounts to manage, add a "Create Community" link.
3. When you click the link, you get a field to specify the name of the community and click OK.
4. The machinery buzzes and churns for a second, creating the account, setting the current user as delegate etc.
5. The community shows up as a new icon in the list of accounts to manage on the manage page.
6. User clicks on new icon to use existing manage function.

**Comment by MrPetovan** [comment]
Yes, it’s “just” that, but as we are a team of volunteers, someone has to come forward and spend their time to implement “just” that. And until someone does, I’m afraid this will stay counter-intuitive.

**Comment by annando** [comment]
Would be great if someone would find some time for it - but I'm the backend guy, frontend is pain for me.

**Comment by divnull** [comment]
Having never seen your code before, what backend methods would such a UI invoke?

**Comment by LubuWest** [comment]
Is this still on the agenda? Another user asked for this feature...

**Comment by rabuzarus** [comment]
I have started some working draft at https://github.com/rabuzarus/friendica/tree/20190204_-_show_forum_account_in_vcard based on the `register` module.
But it isn't polished and I'm running out of time.

Some very important questions remain:
* I habe to say, that I'm overwhelmed with the permissions - next to the the different `Close`, `Approve` and `Open` we have config settings (max_daily_registrations) which have an influence on the register process. The question is: would be the logic the same linke on external registration? When do we have to send an email to the user?
* I think from a GUI point I would like to have 3 sub-pages which guide through the creation process: 1.) select account type 2.) chose your name and nick 3.) should the new page have an entry in the directory? and confirmation with the users password. At the moment I think about submitting the data from each subpage to the `post()` function and provide the the template data through `post()` instead of `content()`. Is this the way to go or are there any better ways to solve this?
* do I need something like this: https://github.com/friendica/friendica/blob/develop/mod/settings.php#L560-L567 on prvgroup creation or is ist just something for later, when the user wants to change the settings?
* my approach requires a user password. I have no idea about OpenID. This needs to implement someone else

**Comment by MrPetovan** [comment]
> The question is: would be the logic the same like on external registration? When do we have to send an email to the user?

I'm not exactly sure what is the actual question so I'll give my opinion on the group/community "user" creation: it should abide by the current registration rules, including:
- Open/close registrations
- Optional required approval
- Registration rate limiting

The only rule I'm not sure is relevant is the invite-only policy.

> Is this the way to go or are there any better ways to solve this?

No, the `content()` method has to be used for all display. The way you would do it is that you would split the `content()` and `post()` method in three separate private methods, one for each step, separated by a GET/POST/SESSION parameter.

> do I need something [like this](https://github.com/friendica/friendica/blob/develop/mod/settings.php#L560-L567) on prvgroup creation or is ist just something for later, when the user wants to change the settings?

I'm not sure what this is for, I'll defer to @annando.
